### DISCUSSION TITLE
Own Config files

### DISCUSSION THREAD

**Discussion by sebska** [john30/ebusd#1166] [Q&A]
Hi there,
i managed to get it working in Home-Assistant (HAOS on a RPI 4) with the addon. I have mananged to use a copy of ebusd-configuration (2.1.x/de) in my /config Folder in HA as well as a local mqtt-hassio.cfg.

Now i want to customize the files in the configuration, but i failed.

I have a buch of values, that show up as "unkown".
My boiler is a Vaillant ecoTEC plus, bai00.

The files loaded are: 08.bai.csv, bai.308523.inc and the inclusions hcmode.inc, errors.inc and both _templates.csv.

Question 1: Now I thought I could simply comment out the lines with the unknown values in file bai.308523.inc and hcmode.inc with "# ", but that doesn't seem to be the case. The error message is:

`2024-01-24 09:22:14.677 [main error] error executing instructions: ERR: invalid numeric argument, last error: error loading "vaillant/bai.308523.inc" for "08": ERR: invalid numeric argument vaillant/bai.308523.inc:9: ERR: invalid numeric argument, id 0D704, included "vaillant/hcmode.inc" for "08"`

Can i just delete them without getting a longer list of unkown messages on the bus (e.g. 2024-01-15 07:18:25.487 [update notice] received unknown MS cmd: 1008b5110101 / 09645000044bff0100ff) ?

Question 2: according to https://github.com/john30/ebusd/wiki/4.1.-Message-definition you can do somthing with the language in the [CONFIG]-filed. I am loading ebusd with --configlang=de, but i would read that i have to change bai.308523.inc as well: in the first line replace 
`# type (r[1-9];w;u),circuit,name,[comment],[QQ],ZZ,PBSB,[ID],field1,part (m/s),datatypes/templates,divider/values,unit,comment` with 
`# type (r[1-9];w;u),circuit,name,[comment.de],[QQ],ZZ,PBSB,[ID],field1,part (m/s),datatypes/templates,divider/values,unit,comment`

Or ist it a misunderstanding?

More questions will follow, i am sure.....

Sebastian

**Comment by JonesGW-MGD** [comment]
Question 1: Not sure exactly what you've done but you can certainly add the hash symbol to comment out lines in bai308253.inc without incurring errors, and the error you posted doesn't indicate an issue with lines you may have commented out in the file to me - have you made any other edits?

Question 2: What do you have in your config path?  Assuming you've git cloned the repository locally, I would have thought if you have this in configpath:

```
/config/ebusd-configuration/latest/de/
``` 
It is going to pick up the hierarchy of files that are in the German language folder without further alteration?

Again - caveat - that I am only recently working with ebusd!

**Comment by JonesGW-MGD** [comment]
In fact I think in line 9 of bai.308253.inc you've made a partial edit of the code - the standard file has:

```
r;wi,,PartloadHcKW,d.00 Heizungsteillast,,,,"0704",,,power,,,Heizungsteillast
``` 

Whereas I'm guessing you have:

```
r;wi,,PartloadHcKW,d.00 Heizungsteillast,,,,"0D704",,,power,,,Heizungsteillast
``` 

which is throwing this error:

```
ERR: invalid numeric argument vaillant/bai.308523.inc:9: ERR: invalid numeric argument, id 0D704
```

**Reply by JonesGW-MGD** [reply]
Moreover - If you find that 0704 returns unknown, try 6C00.  I have a similar but different boiler to you - 0704 returned unknown, whereas 6C00 return a value - in my case 25Kw - this is wrong for my setup (I have a 35Kw combination boiler)  I think as I would expect it to return 'Auto' as I have my 'range rating' / modulation is set to Auto on the panel - but I suspect this is because this code is defined as 'power' whereas it is more likely a string on auto range-rating boilers but I haven't got into this yet....

**Comment by sebska** [comment]
Question 1 solved: my LibreOfficeCalc, which i used to choose for changing the csv and inc Files messed around with the "". They were missing after editing. What tools are you using? Please wirte a comment.

**Reply by Koky05** [reply]
I use Notepad++ or VSCode
![image](https://github.com/john30/ebusd/assets/89223022/fcd98616-53ac-4ead-b576-a9db245506a1)

**Reply by JonesGW-MGD** [reply]
I've used the Studio Server Code add-on within Home Assistant- which is overkill for a simple editing tool, but allows you to edit in-situ and autosaves etc.

**Reply by Koky05** [reply]
@JonesGW-MGD  "Studio Server Code" is it same thing as VSCode. It is always "Microsoft Visual Studio Code". You run it as addon and I as separate docker container on another machine for better access all my source codes and without dependencies on HA (to access HA files I use SSH File extension). Also for work I use VSCode as desktop app.

**Comment by sebska** [comment]
thanks, but back to question number 2:

how does it work with the comment-field in the .csv/.inc and the --configlang=de?

**Reply by Koky05** [reply]
It picks values from .csv/.inc files by preferences where column has suffix same as defined language. But if no definition was found (no suffix in columns name) it ignore language preferences.

**Reply by sebska** [reply]
but where do i have to insert ".de"? in the first line where all definitions of colums are?
Like 

`# type (r[1-9];w;u),circuit,name,[comment.de],[QQ],ZZ,PBSB,[ID],field1,part (m/s),datatypes/templates,divider/values,unit,comment`
